aikau/src/main/resources/alfresco/services/FileUploadService.js

```diff
@@ -0,0 +1,708 @@
+/**
+ * Copyright (C) 2005-2016 Alfresco Software Limited.
+ *
+ * This file is part of Alfresco
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * <p>This service can be used to control the uploading of content as well as
+ * the updating the content of existing nodes on an Alfresco Repository.</p>
+ * 
+ * @module alfresco/services/FileUploadService
+ * @extends module:alfresco/services/BaseService
```

Comment:
OK, I've started comparing and I can see that what you've done is change some of the code, re-order and give them what you think are better function names - however, the core functionality remains the same - I don't want to have different code doing exactly the same thing in two different files.

We can't change the names, but people aren't going to be seeing them - I took them from the original YUI upload - whilst they might not be the best names - they're only names.

I'd like you to go back through and abstract the common code to a mixin - I'm sure that the actual code changes you've made are fine but I'll verify with some testing in place on Share on Monday.